authserver/api/role.py

```diff
@@ -128,6 +127,90 @@ def update(self, id: str, partial=True):
             return self.response_handler.exception_response(exception_name, request=request_data)
 
 
+class AuthorizedScopeResource(Resource):
+    """Authorized scope resource for linking roles to scopes. """
+
+    def __init__(self):
+        self.authorized_scope_schema = AuthorizedScopeSchema()
+        self.authorized_scopes_schema = AuthorizedScopeSchema(many=True)
+
+        self.response_handler = ResponseBody()
+
+    @require_oauth()
+    def get(self, id: str = None, sid: str = None):
+        if sid is None:
+            try:
+                authorized_scopes = AuthorizedScope.query.all()
+                scopes_obj = self.authorized_scopes_schema.dump(authorized_scopes)
+                scopes_obj_clean = [{k: v for k, v in authorized_scope.items() if k != 'scope_id' and k != 'role_id' and k != 'role'} for authorized_scope in scopes_obj]
+                return self.response_handler.get_all_response(scopes_obj_clean)
+            except Exception:
+                return self.response_handler.exception_response('Unknown')
+        else:
+            try:
+                authorized_scope = AuthorizedScope.query.filter(
+                    AuthorizedScope.role_id == id, AuthorizedScope.scope_id == sid).first()
+                scope_obj = self.authorized_scope_schema.dump(authorized_scope)
+                scope_obj.pop('role_id', None)
+                scope_obj.pop('scope_id', None)
```

Comment:
I would move lines 154-155 under the `if scope_obj`, so you do not call `pop` on a null/empty object. So, this:

```python
if scope_obj:
    scope_obj.pop('role_id', None)
    scope_obj.pop('scope_id', None)
    return self.response_handler.get_one_response(scope_obj)
```

Reply:
Good call.